### PR TITLE
Sai interface

### PR DESCRIPTION
**Pull request by jasonstack**

### CODE REVIEW

File: src/java/org/apache/cassandra/index/IndexRegistry.java [hunk 1]
```diff
@@ -189,7 +203,50 @@ public Searcher searcherFor(ReadCommand command)
             }
         };
 
-        public void registerIndex(Index index)
+        Index.Group group = new Index.Group() {
```
Comment (mike-tr-adamson) [inline src/java/org/apache/cassandra/index/IndexRegistry.java:206]: nit: formatting

File: src/java/org/apache/cassandra/index/IndexRegistry.java [hunk 2]
```diff
@@ -189,7 +203,50 @@ public Searcher searcherFor(ReadCommand command)
...
+            }
+
+            @Override
+            public boolean containsIndex(Index i)
+            {
+                return index == i;
+            }
+
+            @Override
+            public Index.Indexer indexerFor(Predicate<Index> indexSelector, DecoratedKey key, RegularAndStaticColumns columns, int nowInSec, WriteContext ctx, IndexTransaction.Type transactionType, Memtable memtable)
```
Comment (mike-tr-adamson) [inline src/java/org/apache/cassandra/index/IndexRegistry.java:230]: Should this and `getFlushObserver` be `Nullable` as well?

**Comment by jasonstack** [comment]
created a new PR pointing to datastax repo: https://github.com/datastax/cassandra/pull/24

File: src/java/org/apache/cassandra/index/IndexRegistry.java (hunk 1 above)
Comment (jasonstack) [reply src/java/org/apache/cassandra/index/IndexRegistry.java:206]: +1

File: src/java/org/apache/cassandra/index/IndexRegistry.java (hunk 2 above)
Comment (jasonstack) [reply src/java/org/apache/cassandra/index/IndexRegistry.java:230]: +1

**Pull request closed without merging**